src/rgw/rgw_lc.cc

```diff
@@ -43,6 +55,49 @@ void RGWLifecycleConfiguration::_add_rule(LCRule *rule)
   prefix_map[rule->get_prefix()] = rule->get_expiration().get_days();
 }
 
+bool RGWLifecycleConfiguration::check_and_add_rule(LCRule *rule)
```

Comment:
The add_rule() is used in parsing the xml (RGWLifecycleConfiguration_S3::xml_end). If we change it, we will check the rule twice.